url7606378bcc_ljy2010a_protobuf2_tgz-pJ8-com_game_gameObj_bufBuilder_ReflectBuilderJ8/src/com/game/protobuf/GameProto.java

```diff
@@ -0,0 +1,7112 @@
+    com.game.protobuf.GameProto.UserInfo getUserInfo();
+    com.game.protobuf.GameProto.UserInfoOrBuilder getUserInfoOrBuilder();
+  }
+  public static final class LoginRequest extends
+      com.google.protobuf.GeneratedMessage
+      implements LoginRequestOrBuilder {
+    // Use LoginRequest.newBuilder() to construct.
+    private LoginRequest(Builder builder) {
+      super(builder);
+    }
+    private LoginRequest(boolean noInit) {}
+    
+    private static final LoginRequest defaultInstance;
+    public static LoginRequest getDefaultInstance() {
+      return defaultInstance;
+    }
+    
+    public LoginRequest getDefaultInstanceForType() {
+      return defaultInstance;
+    }
+    
+    public static final com.google.protobuf.Descriptors.Descriptor
+        getDescriptor() {
+      return com.game.protobuf.GameProto.internal_static_protobuf_LoginRequest_descriptor;
+    }
+    
+    protected com.google.protobuf.GeneratedMessage.FieldAccessorTable
+        internalGetFieldAccessorTable() {
+      return com.game.protobuf.GameProto.internal_static_protobuf_LoginRequest_fieldAccessorTable;
+    }
+    
+    private int bitField0_;
+    // optional .protobuf.UserInfo userInfo = 1;
+    public static final int USERINFO_FIELD_NUMBER = 1;
+    private com.game.protobuf.GameProto.UserInfo userInfo_;
+    public boolean hasUserInfo() {
+      return ((bitField0_ & 0x00000001) == 0x00000001);
+    }
+    public com.game.protobuf.GameProto.UserInfo getUserInfo() {
+      return userInfo_;
+    }
+    public com.game.protobuf.GameProto.UserInfoOrBuilder getUserInfoOrBuilder() {
+      return userInfo_;
+    }
+    
+    private void initFields() {
+      userInfo_ = com.game.protobuf.GameProto.UserInfo.getDefaultInstance();
+    }
+    private byte memoizedIsInitialized = -1;
+    public final boolean isInitialized() {
+      byte isInitialized = memoizedIsInitialized;
+      if (isInitialized != -1) return isInitialized == 1;
+      
+      if (hasUserInfo()) {
+        if (!getUserInfo().isInitialized()) {
+          memoizedIsInitialized = 0;
+          return false;
+        }
+      }
+      memoizedIsInitialized = 1;
+      return true;
+    }
+    
+    public void writeTo(com.google.protobuf.CodedOutputStream output)
+                        throws java.io.IOException {
+      getSerializedSize();
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
+        output.writeMessage(1, userInfo_);
+      }
+      getUnknownFields().writeTo(output);
+    }
+    
+    private int memoizedSerializedSize = -1;
+    public int getSerializedSize() {
+      int size = memoizedSerializedSize;
+      if (size != -1) return size;
+    
+      size = 0;
+      if (((bitField0_ & 0x00000001) == 0x00000001)) {
+        size += com.google.protobuf.CodedOutputStream
+          .computeMessageSize(1, userInfo_);
+      }
+      size += getUnknownFields().getSerializedSize();
+      memoizedSerializedSize = size;
+      return size;
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 2951:2961, 3387:3397, 3823:3833.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.